tidy3d/plugins/autograd/functions.py

```diff
@@ -161,7 +162,7 @@ def pad(
     array: np.ndarray,
     pad_width: Union[int, Tuple[int, int]],
     *,
-    mode: _pad_modes = "constant",
+    mode: PaddingType = "constant",
```

Comment:
maybe we should start using Enums for these kinds of things. There's an open issue for it but nobody's found a great solution yet.

https://github.com/flexcompute/tidy3d/issues/564

Reply:
Agreed. So something like
```python
class PaddingType(StrEnum):
    CONIC = "conic"

def f(p: PaddingType):
    ...

f(PaddingType.CONIC)
```